{{cookiecutter.project_name}}/tasks.py

```diff
@@ -200,59 +237,71 @@ def pre_commit(c: Context, auto_fix: bool):
     if ("fixed" in result.stdout or "reformatted" in result.stdout) and auto_fix:
         _add_commit(c, msg="style: Auto-fixes from pre-commit")
 
-        print(f"{Msg.DOING} Fixed errors, re-running pre-commit checks")
+        print(f"{msg_type.DOING} Fixed errors, re-running pre-commit checks")
         second_result = c.run(pre_commit_cmd, pty=NOT_WINDOWS, warn=True)
         exit_if_remaining_errors(second_result)
     else:
         if result.return_code != 0:
-            print(f"{Msg.FAIL} Pre-commit checks failed")
+            print(f"{msg_type.FAIL} Pre-commit checks failed")
             exit(1)
 
 
-def mypy(c: Context):
-    echo_header(f"{Msg.CLEAN} Running mypy")
-    c.run("mypy .", pty=NOT_WINDOWS)
+@task
+def static_type_checks(c: Context):
+    echo_header(f"{msg_type.CLEAN} Running static type checks")
+    c.run("pyright .", pty=NOT_WINDOWS)
 
 
 @task
-def install(c: Context, args: str = "", msg: bool = True):
+def install(c: Context, pip_args: str = "", msg: bool = True):
     """Install the project in editable mode using pip install"""
     if msg:
-        echo_header(f"{Msg.DOING} Installing project")
+        echo_header(f"{msg_type.DOING} Installing project")
 
     if NOT_WINDOWS:
-        c.run(f"pip install -e '.[dev,tests,docs]' {args}")
+        c.run(f"pip install -e '.[dev,tests,docs]' {pip_args}")
     else:
-        c.run(f"pip install -e .[dev,tests,docs] {args}")
+        c.run(f"pip install -e .[dev,tests,docs] {pip_args}")
 
 
 @task
-def setup(c: Context, python_version: str = "3.9"):
-    """Confirm that a git repo exists and setup a virtual environment."""
+def setup(c: Context, python: Optional[str] = None):
+    """Confirm that a git repo exists and setup a virtual environment.
+
+    Args:
+        c: Invoke context
+        python: Path to the python executable to use for the virtual environment.
+    """
     git_init(c)
 
-    venv_name = setup_venv(c, python_version=python_version)
+    if python is None:
+        # get path to python executable
+        python = shutil.which("python")
```

Comment:
Would love if this could default to 3.9 if present; currently will use 3.8 on most MacOS versions. Will try to come up with something.

Reply:
Yea would like so too. We could have it read an OS path setting (as it is user-specific and not project specific)

Reply:
Yeah, agreed! For main, I've added that it attempts `python{version_number}` and, if not found, looks for `which python`. Super open to improvements above that if possible 👍

Reply:
Seems like. I would probably prefer if it simply allowed for python_path to also be a version and then simply tries the path, then attempts the version and if it fails raise an error. Not a big thing though.

Reply:
Yeah, that'd be even better! Open to a PR 👍